rclcpp/include/rclcpp/executor.hpp

```diff
@@ -244,9 +245,16 @@ class Executor
     }
     std::chrono::nanoseconds timeout_left = timeout_ns;
 
-    while (rclcpp::ok(this->context_)) {
+    if (spinning.exchange(true)) {
+      throw std::runtime_error("spin_some() called while already spinning");
```

Comment:
This error message doesn't seem right. What does `spin_some()` have to do with this function?

Reply:
I change the error message. Thanks.